app/src/androidTest/java/org/mozilla/fenix/ui/SponsoredShortcutsTest.kt

```diff
@@ -0,0 +1,298 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.customannotations.SmokeTest
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.Constants.defaultTopSitesList
+import org.mozilla.fenix.helpers.HomeActivityIntentTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.helpers.TestHelper.exitMenu
+import org.mozilla.fenix.helpers.TestHelper.getSponsoredShortcutTitle
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ * Tests Sponsored shortcuts functionality
+ */
+
+class SponsoredShortcutsTest {
+    private lateinit var mDevice: UiDevice
+    private lateinit var mockWebServer: MockWebServer
+    private val defaultSearchEngine = "Amazon.com"
+
+    @get:Rule
+    val activityIntentTestRule = HomeActivityIntentTestRule.withDefaultSettingsOverrides(skipOnboarding = true)
+
+    @Before
+    fun setUp() {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    // Expected for en-us defaults
+    @SmokeTest
+    @Test
+    fun verifySponsoredShortcutsListTest() {
+        homeScreen {
+            defaultTopSitesList.values.forEach { value ->
+                verifyExistingTopSitesTabs(value)
+            }
+        }.openThreeDotMenu {
+        }.openCustomizeHome {
+            verifySponsoredShortcutsCheckBox(true)
+            clickSponsoredShortcuts()
+            verifySponsoredShortcutsCheckBox(false)
+        }.goBack {
+            verifyNotExistingSponsoredTopSitesList()
+        }
+    }
+
+    @Test
+    fun openSponsoredShortcutTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openSponsoredShortcut(sponsoredShortcutTitle) {
+            verifyUrl(sponsoredShortcutTitle)
+        }
+    }
+
+    @Test
+    fun openSponsoredShortcutInPrivateBrowsingTest() {
+        var sponsoredShortcutTitle = ""
+
+        homeScreen {
+            sponsoredShortcutTitle = getSponsoredShortcutTitle(2)
+        }.openContextMenuOnSponsoredShortcut(sponsoredShortcutTitle) {
+        }.openTopSiteInPrivateTab {
+            verifyUrl(sponsoredShortcutTitle)
+        }
+    }
+
+    @Ignore("Failing, see: https://github.com/mozilla-mobile/fenix/issues/25926")
+    @Test
+    fun verifySponsoredShortcutsSponsorsAndPrivacyOptionTest() {
```

Comment:
a small change here, since it's relevant from the button's title what it does, you can shorten the title to: verifySponsorsAndPrivacyLinkTest

Reply:
Done !